packages/module/src/WarningModal/WarningModal.tsx

```diff
@@ -13,11 +13,17 @@ export interface WarningModalProps extends Omit<ModalProps, 'ref'> {
   /** Whether modal requires a checkbox before confirming */
   withCheckbox?: boolean;
   /** Custom checkbox label */
-  checkboxLabel?: string;
+  checkboxLabel?: ReactNode;
   /** Visual variant of the confirm button */
   confirmButtonVariant?: ButtonVariant;
   /** Custom OUIA ID */
   ouiaId?: string | number;
+  /** Whether modal requires a text confirmation */
+  textConfirmation?: TextInputProps;
+  /** Label for the text confirmation */
+  textConfirmationLabel?: (deleteName?: string) => ReactNode;
+  /** Text the user should type to confirm selection when using textConfirmation */
+  deleteName?: string;
```

Comment:
```suggestion
  /** Confirmation text input props */
  confirmationInputProps?: TextInputProps;
  /** Label for the text confirmation input */
  confirmationInputLabel?: (deleteName?: string) => ReactNode;
  /** Text the user should type to confirm selection when using confirmation input */
  confirmationText?: string;
```

how about renaming the props like so? Mabe it's a little bit more descriptive